src/agent/agent_info/CMakeLists.txt

```diff
@@ -12,8 +12,11 @@ find_package(Boost REQUIRED COMPONENTS uuid)
 find_package(nlohmann_json CONFIG REQUIRED)
 
 add_library(AgentInfo src/agent_info.cpp src/agent_info_persistance.cpp)
-target_include_directories(AgentInfo PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
-target_link_libraries(AgentInfo PUBLIC nlohmann_json::nlohmann_json PRIVATE SQLiteManager Boost::uuid Logger)
+target_include_directories(AgentInfo PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+    ${CMAKE_SOURCE_DIR}/common/data_provider/include
```

Comment:
I'll make this PRIVATE for now, but let's keep this in mind for future improvements.

Reply:
It should be pretty simple, just pass a function that sets the parameters you want to set, then the Agent class, which owns AgentInfo will set this lambda, it is less problematic that the Agent class links to sysInfo since it's the top most target.